tests/datetime_test.py

```diff
@@ -241,9 +241,10 @@ def test_datetime_repr_not_unpicklable(self):
         obj = datetime.datetime.now()
         pickler = jsonpickle.pickler.Pickler(unpicklable=False)
         flattened = pickler.flatten(obj)
-        self.assertFalse(tags.REPR in flattened)
-        self.assertFalse(tags.OBJECT in flattened)
-        self.assertEqual(obj.isoformat(), flattened)
+        assert tags.REPR not in flattened
```

Comment:
Is there a reason not to use the `self.assertBoolean` functions? I would think it'd be easier as it would allow us to easily modify behavior of the functions if needed for whatever reason someday, plus for consistency with the rest of the existing tests.

Reply:
Yeah, there's some logic to this, though it will look inconsistent from the perspective of the individual diffs.

The older camelCased `assertBoolean` methods are from the old days when the `unittest` module was new and the names were chosen to match the exact names of the Java junit library, which exposed a similar API.

One of pytest's selling points is that it lets us use a consistent naming convention by by leveraging the language's existing `assert` statements. pytest augments `assert` statements by adding pretty printing and diffing, which makes the advantages of using the junit-inspired methods almost non-existent. It does drop the ability to extend these methods, but I hope we won't need to do that :crossed_fingers: .

So over time I've taken the approach of replacing older tests' assertBoolean methods with pytest-native equivalents as I run into them updating the test method for some other reason. I'll just do this incrementally as tests get updated, though at some point I might make a series of file-by-file commits to updates the test suite all at once and leverage more of pytest's niceties. Until then the diffs will look like they're inconsistent with their surroundings, but it's okay since it should be incrementally going towards the newer pytest style.

While I find that `asserts` are shorter, simpler, and easier to read/type, it's mostly aesthetic as both ultimately do the same thing. The python core team decided that they'll keep the junit names around forever so there's not any pressing deprecation or technical reason to change it on any timeframe.

Reply:
Ah, that makes sense, thanks! I forgot that we use pytest, and from the POV of pytest it makes total sense to use asserts and not the unittest functions.